src/pkg/utils/io.go

```diff
@@ -471,9 +471,20 @@ func IsTextFile(path string) (bool, error) {
 	}
 	defer f.Close() // Make sure to close the file when we're done
 
-	// Read the first 512 bytes of the file
+	// Get file stat
+	stat, err := f.Stat()
+	if err != nil {
+		return false, err
+	}
+
+	// Create 512 byte buffer
 	data := make([]byte, 512)
-	n, err := f.Read(data)
```

Comment:
We will still want to read the first bytes as well in case text is added to the end of a binary file (also the end of a file is unlikely to have BOMs or the starter sequences to get an accurate mimetype - this function doesn't _really_ need that but it would be nicer to base the primary decision information on whether this is a text file off of a more accurate mimetype).

Reply:
My only question to that is are you fine with it always reading front and back? Since not matter what, due to this edge case with installers like NVIDIA, it will always need to do that. I guess in order to pass the "is this a text file" check is both front and back must be a text mimetype. Not hard to do at all, I can make this change either today or early next week.